### PR TITLE
added scorecard action for OSSF

### PR DESCRIPTION
**Pull request by VedRatan**
This PR adds the OSSF scorecard action, which will help to analyze the codebase of Nephio to make sure that the codebase is complaint with OSSF best practices.

### CODE REVIEW

File: .github/workflows/openssf_scorecard.yaml
Comment (nyrahul) [inline .github/workflows/openssf_scorecard.yaml:32]:
```suggestion
          # repo_token: ${{ secrets.SCORECARD_READ_TOKEN }}
```
`repo_token` is only needed if Branch-Protection check needs to be enabled. We can do without this since maintaining this token might have operational issues.

**Review by nyrahul — CHANGES_REQUESTED** [review]

**Comment by nyrahul** [comment]
Hey @VedRatan , can you please rebase this? Thanks

**Review by nyrahul — APPROVED** [review]
LGTM

**Comment by liamfallon** [comment]
I don't know why this has not been merged, it has lgtm and approval from @henderiw and gltm from @johnbelamaric

**Comment by johnbelamaric** [comment]
Don't know either - but I merged it manually

**Comment by liamfallon** [comment]
> Don't know either - but I merged it manually

curiouser and curiouser :-)

**Comment by johnbelamaric** [comment]
I think something is wrong with prow and/or tide. I really don't think we should have to do merge commits to be at the tip of main before merging. I think tide is supposed to do that. I think it also does these merges. Not sure though. @radoslawc any ideas?

**Comment by radoslawc** [comment]
@johnbelamaric tide is stuck on a merge in nephio-example-packages, but I cannot seem to find PR responsible for it with what information tide provides in logs:
```
  "msg": "Merge failed.",
  "out": "Auto-merging ueransim/ue/ue-deployment.yaml\nCONFLICT (content): Merge conflict in ueransim/ue/ue-deployment.yaml\nAuto-merging ueransim/ue/ue-configmap.yaml\nCONFLICT (content): Merge conflict in ueransim/ue/ue-configmap.yaml\nAuto-merging ueransim/namespace.yaml\nAuto-merging ueransim/gnb/service_gnb-service.yaml\nCONFLICT (content): Merge conflict in ueransim/gnb/service_gnb-service.yaml\nCONFLICT (modify/delete): ueransim/gnb/gnb-n3-nad.yaml deleted in HEAD and modified in edf3f2ffdbd5469da058a5d037e4d56f2d5fcdd1. Version edf3f2ffdbd5469da058a5d037e4d56f2d5fcdd1 of ueransim/gnb/gnb-n3-nad.yaml left in tree.\nCONFLICT (modify/delete): ueransim/gnb/gnb-n2-nad.yaml deleted in HEAD and modified in edf3f2ffdbd5469da058a5d037e4d56f2d5fcdd1. Version edf3f2ffdbd5469da058a5d037e4d56f2d5fcdd1 of ueransim/gnb/gnb-n2-nad.yaml left in tree.\nAuto-merging ueransim/gnb/gnb-deployment.yaml\nCONFLICT (content): Merge conflict in ueransim/gnb/gnb-deployment.yaml\nCONFLICT (modify/delete): ueransim/gnb/gnb-configmap.yaml deleted in HEAD and modified in edf3f2ffdbd5469da058a5d037e4d56f2d5fcdd1. Version edf3f2ffdbd5469da058a5d037e4d56f2d5fcdd1 of ueransim/gnb/gnb-configmap.yaml left in tree.\nAuto-merging resource-backend/crd/bases/vlan.resource.nephio.org_vlans.yaml\nAuto-merging resource-backend/crd/bases/vlan.resource.nephio.org_vlanindices.yaml\nAuto-merging resource-backend/crd/bases/vlan.resource.nephio.org_vlanclaims.yaml\nAuto-merging resource-backend/crd/bases/ipam.resource.nephio.org_networkinstances.yaml\nAuto-merging resource-backend/crd/bases/ipam.resource.nephio.org_ipprefixes.yaml\nAuto-merging resource-backend/crd/bases/ipam.resource.nephio.org_ipclaims.yaml\nAuto-merging resource-backend/app/controller/grpc/service-grpc.yaml\nAuto-merging resource-backend/app/controller/deployment-controller.yaml\nAuto-merging resource-backend/app/controller/clusterrole-controller.yaml\nAuto-merging repository/token-configsync.yaml\nAuto-merging repository/set-values.yaml\nAuto-merging porch-dev/0-repositories.yaml\nAuto-merging porch-dev/0-packagevariantsets.yaml\nAuto-merging porch-dev/0-packagevariants.yaml\nAuto-merging porch-dev/0-packagerevs.yaml\nAuto-merging nephio-workload-cluster/workload-cluster.yaml\nAuto-merging nephio-workload-cluster/pv-rootsync.yaml\nAuto-merging nephio-workload-cluster/pv-repo.yaml\nAuto-merging nephio-workload-cluster/pv-kindnet.yaml\nAuto-merging nephio-workload-cluster/pv-configsync.yaml\nAuto-merging nephio-workload-cluster/pv-cluster.yaml\nAuto-merging nephio-controllers/crd/bases/infra.nephio.org_workloadclusters.yaml\nAuto-merging nephio-controllers/app/controller/deployment-controller.yaml\nAuto-merging configsync/config-management-operator.yaml\nAuto-merging cluster-capi-kind/cluster.yaml\nSquash commit -- not updating HEAD\nAutomatic merge failed; fix conflicts and then commit the result.\n",
```
Any ideas where is this change coming from?

**Comment by johnbelamaric** [comment]
I searched for the commit listed in there:

https://github.com/search?q=repo%3Anephio-project%2Fnephio-example-packages+edf3f2ffdbd5469da058a5d037e4d56f2d5fcdd1&type=pullrequests

It's from this PR: https://github.com/nephio-project/nephio-example-packages/pull/43

**Comment by johnbelamaric** [comment]
But that's reflected in GH. I don't think it should *stop* tide from working.

But this is more what I would *expect* to see - the merge would fail if tide couldn't rebase without conflict. Instead, right now across all of our repos, we have to rebase manually.

**Comment by radoslawc** [comment]
I've updated PR branch with main and https://github.com/nephio-project/nephio-example-packages/pull/79 has been merged automatically by tide, and it was hanging there as well, let's see if that unclogged tide.

**Comment by radoslawc** [comment]
> But that's reflected in GH. I don't think it should _stop_ tide from working.
> 
> But this is more what I would _expect_ to see - the merge would fail if tide couldn't rebase without conflict. Instead, right now across all of our repos, we have to rebase manually.

@johnbelamaric is 'Require branches to be up to date before merging' enabled on this repo? Seems like it's causing rebase issue with tide. From prow's docs:

Do not enable the “Require branches to be up to date before merging” GitHub setting for repos managed by Tide. This requires all PRs to be rebased before merge so that PRs are always simple fast-forwards. This is a simplistic way to ensure that PRs are tested against the most recent base branch commit, but Tide already provides this guarantee through a more sophisticated mechanism that does not force PR authors to rebase their PR whenever another PR merges first. Enabling this GH setting may cause unexpected Tide behavior, provides absolutely no benefit over Tide’s natural behavior, and forces PR author’s to needlessly rebase their PRs. Don’t use it on Tide managed repos.

**Pull request merged**
